src/LoginPage/components/RegisterForm.js

```diff
@@ -111,27 +112,38 @@ class RegisterForm extends Component {
     };
   }
 
-  storeProfilePicture() {
+  storeProfilePicture () {
     const ext = this.state.img.name.slice(this.state.img.name.lastIndexOf('.'));
     return firebase.storage()
       .ref(`/users/${this.state.user.first_name}${this.state.user.last_name}${ext}`)
       .put(this.state.img)
       .then(snapshot => {
         this.state.user.photo = snapshot.downloadURL;
       });
-  }
-
+    };   
+  
   onSubmit(e) {
-    this.storeProfilePicture()
-      .then(() => {
-        let user = this.state.user
-        axios.post("http://localhost:9000/api/v1/users", user).then(payload => {
-          this.setState({ userMade: true });
-        }).catch(err => {
-          alert(err)
-        });
-      });
-  }
+  if(this.state.imgURL.startsWith("data:image")){
+   this.storeProfilePicture()
+     .then(() => {
+       let user = this.state.user
+       axios.post("http://localhost:9000/api/v1/users", user).then(payload => {
+         this.setState({ userMade: true });
+       }).catch(err => {
+         alert(err)
+       });
+     });
+   }
+   else
+   {
+     let user = this.state.user
+     axios.post("http://localhost:9000/api/v1/users", user).then(payload => {
```

Comment:
Same thing here. Don't use the hard coded url. Because the api won't always point to localhost:9000, so we store that part of the url in a variable that we can change on demo and production environments